-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f6804c/13265/summary.html
**COMMIT**: e61a5cefd43edd09874f6720049c245ba71875c9
**CMSSW**: CMSSW_11_3_X_2021-03-03-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33066/13265/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-03-2300/src/FWCore/PyDevParameterSet/test/makepset_t.cppunit.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-03-03-2300/src/FWCore/PyDevParameterSet/test/readpsetsfrom_t.cppunit.cc
>> Building binary testPyBind11ParameterSet
/cvmfs/cms-ib.cern.ch/nweek-02670/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc900/src/FWCore/PyDevParameterSet/test/testPyBind11ParameterSet/makepset_t.cppunit.cc.o: in function `testmakepset::fileinpathAux()':
makepset_t.cppunit.cc:(.text+0x24aa): undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
collect2: error: ld returned 1 exit status
>> Deleted: tmp/slc7_amd64_gcc900/src/FWCore/PyDevParameterSet/test/testPyBind11ParameterSet/testPyBind11ParameterSet
gmake: *** [tmp/slc7_amd64_gcc900/src/FWCore/PyDevParameterSet/test/testPyBind11ParameterSet/testPyBind11ParameterSet] Error 1
>> Entering Package FWCore/PrescaleService
>> Leaving Package FWCore/PrescaleService
>> Package FWCore/PrescaleService built
</pre>

--- a/DQMServices/Components/src/fillJson.cc
+++ b/DQMServices/Components/src/fillJson.cc
@@ -13,11 +13,11 @@
 // system include files
 #include <boost/property_tree/json_parser.hpp>
 #include <boost/property_tree/ptree.hpp>
-#include <boost/filesystem.hpp>
 #include <boost/format.hpp>
 
 #include <string>
 #include <sstream>
+#include <filesystem>
 
 // user include files
 #include "FWCore/ServiceRegistry/interface/Service.h"
@@ -33,7 +33,6 @@ boost::property_tree::ptree dqmfilesaver::fillJson(int run,
                                                    const std::string& mergeTypeStr,
                                                    evf::FastMonitoringService* fms) {
   namespace bpt = boost::property_tree;
-  namespace bfs = boost::filesystem;
 
   bpt::ptree pt;
 
@@ -57,7 +56,7 @@ boost::property_tree::ptree dqmfilesaver::fillJson(int run,
     if (stat(dataFilePathName.c_str(), &dataFileStat) != 0)
       throw cms::Exception("fillJson") << "Internal error, cannot get data file: " << dataFilePathName;
     // Extract only the data file name from the full path
-    dataFileName = bfs::path(dataFilePathName).filename().string();
+    dataFileName = std::filesystem::path(dataFilePathName).filename().string();
   }
   // The availability test of the FastMonitoringService was done in the ctor.
   bpt::ptree data;
@@ -95,7 +94,7 @@ boost::property_tree::ptree dqmfilesaver::fillJson(int run,
     pt.put("definition", "/fakeDefinition.jsn");
   } else {
     // The availability test of the EvFDaqDirector Service was done in the ctor.
-    bfs::path outJsonDefName{
+    std::filesystem::path outJsonDefName{
         edm::Service<evf::EvFDaqDirector>()->baseRunDir()};  //we assume this file is written bu the EvF Output module
     outJsonDefName /= (std::string("output_") + oss_pid.str() + std::string(".jsd"));
     pt.put("definition", outJsonDefName.string());

--- a/FWCore/Catalog/test/FileLocator_t.cpp
+++ b/FWCore/Catalog/test/FileLocator_t.cpp
@@ -2,7 +2,7 @@
 #include "FWCore/Catalog/interface/SiteLocalConfig.h"
 #include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/ServiceRegistry/interface/ServiceRegistry.h"
-#include <boost/filesystem.hpp>
+#include <filesystem>
 
 #include <string>
 
@@ -46,7 +46,7 @@ TEST_CASE("FileLocator", "[filelocator]") {
   std::string CMSSW_BASE(std::getenv("CMSSW_BASE"));
   std::string CMSSW_RELEASE_BASE(std::getenv("CMSSW_RELEASE_BASE"));
   std::string file_name("/src/FWCore/Catalog/test/simple_catalog.xml");
-  std::string full_file_name = boost::filesystem::exists((CMSSW_BASE + file_name).c_str())
+  std::string full_file_name = std::filesystem::exists((CMSSW_BASE + file_name).c_str())
                                    ? CMSSW_BASE + file_name
                                    : CMSSW_RELEASE_BASE + file_name;
 
@@ -84,7 +84,7 @@ TEST_CASE("FileLocator", "[filelocator]") {
     edm::ServiceRegistry::Operate operate(tempToken);
 
     std::string override_file_name("/src/FWCore/Catalog/test/override_catalog.xml");
-    std::string override_full_file_name = boost::filesystem::exists((CMSSW_BASE + override_file_name).c_str())
+    std::string override_full_file_name = std::filesystem::exists((CMSSW_BASE + override_file_name).c_str())
                                               ? CMSSW_BASE + override_file_name
                                               : CMSSW_RELEASE_BASE + override_file_name;
 

--- a/FWCore/PyDevParameterSet/test/makepset_t.cppunit.cc
+++ b/FWCore/PyDevParameterSet/test/makepset_t.cppunit.cc
@@ -21,7 +21,7 @@
 #include <stdlib.h>  // for setenv; <cstdlib> is likely to fail
 #include <string>
 #include <unistd.h>
-#include <boost/filesystem.hpp>
+#include <filesystem>
 
 class testmakepset : public CppUnit::TestFixture {
   CPPUNIT_TEST_SUITE(testmakepset);
@@ -240,7 +240,7 @@ void testmakepset::fileinpathAux() {
       std::string const src("/src");
       std::string local = localBase + src;
       std::string localFile = local + "/Geometry/TrackerSimData/data/trackersens.xml";
-      if (!boost::filesystem::exists(localFile))
+      if (!std::filesystem::exists(localFile))
         CPPUNIT_ASSERT(topo.location() != edm::FileInPath::Local);
       else
         std::cerr << "Disabling test against local path for trackersens.xml as package is checked out in this test"

--- a/FWCore/PythonParameterSet/test/makepset_t.cppunit.cc
+++ b/FWCore/PythonParameterSet/test/makepset_t.cppunit.cc
@@ -21,7 +21,7 @@
 #include <stdlib.h>  // for setenv; <cstdlib> is likely to fail
 #include <string>
 #include <unistd.h>
-#include <boost/filesystem.hpp>
+#include <filesystem>
 
 class testmakepset : public CppUnit::TestFixture {
   CPPUNIT_TEST_SUITE(testmakepset);
@@ -240,7 +240,7 @@ void testmakepset::fileinpathAux() {
       std::string const src("/src");
       std::string local = localBase + src;
       std::string localFile = local + "/Geometry/TrackerSimData/data/trackersens.xml";
-      if (!boost::filesystem::exists(localFile))
+      if (!std::filesystem::exists(localFile))
         CPPUNIT_ASSERT(topo.location() != edm::FileInPath::Local);
       else
         std::cerr << "Disabling test against local path for trackersens.xml as package is checked out in this test"

